kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java

```diff
@@ -92,10 +92,12 @@ public static String getKompiledStringAndWriteSyntaxMacros(ModuleToKORE converte
         return semantics.toString();
     }
 
-    public static Module getKompiledModule(Module mainModule) {
+    public static Module getKompiledModule(Module mainModule, boolean hasAnd) {
         mainModule = new GenerateSortPredicateRules(true).gen(mainModule);
         mainModule = ModuleTransformer.fromSentenceTransformer(new AddSortInjections(mainModule)::addInjections, "Add sort injections").apply(mainModule);
-        mainModule = ModuleTransformer.fromSentenceTransformer(new MinimizeTermConstruction(mainModule)::resolve, "Minimize term construction").apply(mainModule);
+        if (hasAnd) {
+          mainModule = ModuleTransformer.fromSentenceTransformer(new MinimizeTermConstruction(mainModule)::resolve, "Minimize term construction").apply(mainModule);
+        }
```

Comment:
Can you write some description of `hasAnd`?
This is not very descriptive.

Reply:
Sure.